src/backend/src/services/reimbursement-requests.services.ts

```diff
@@ -130,6 +130,8 @@ export default class ReimbursementRequestService {
 
     if (!expenseType) throw new NotFoundException('Expense Type', expenseTypeId);
 
+    if (!expenseType.allowed) throw new HttpException(400, `The expense type ${expenseType} is not allowed!`);
```

Comment:
One quick change. `expenseType` is the entire object. The name is at `expenseType.name`. The error message will be cleaner if it's `The expense type ${expenseType.name} is not allowed!`